src/Security/Listener/LogoutEventListener.php

```diff
@@ -63,8 +62,10 @@ public function __construct(
         $this->serializerResponder = $serializerResponder;
     }
 
-    public function onLogoutSuccess(Request $request): JsonResponse
+    public function onSymfonyComponentSecurityHttpEventLogoutEvent(LogoutEvent $event): void
```

Comment:
Also, not really related to the PR, but it seems the token is not being properly invalidated:

I've tried on POSTMAN:
- Login
- Get assignments - worked fine
- Logout
- Get assignments - still working after logout

This shouldn't be the expect behavior, right?

Reply:
It is because the remove only the refresh token upon logout, but it seems weird. Any idea why did we do it like this? I believe we should remove the access token as well. What do you think guys?

Reply:
1st question:
> Do we need this long name? Why not just onLogoutEvent?
According to "core" behaviour it can be either `onSymfonyComponentSecurityHttpEventLogoutEvent` or `__invoke`. If I put smth like `onAsdEvent`, it will return to me the error
```json
{
    "error": {
        "message": "Failed to create closure from callable: class 'OAT\\SimpleRoster\\Security\\Listener\\LogoutEventListener' does not have a method 'onSymfonyComponentSecurityHttpEventLogoutEvent'",
...
...
...
```
For me it doesn't really matter to have `__invoke` or this "long and fancy" name. If you guys wish I can easily replace it with invoke. @Skyld , wdyt?

Reply:
It's ok as it is 👍  But I am still confused about the missing access token invalidations. There must be a reason for it, but I cannot remember and it looks weird. After logout you should not be able to use your access token imho.

Reply:
now 2nd question, and it's more interesting:
> Also, not really related to the PR, but it seems the token is not being properly invalidated:
I've tried on POSTMAN:
Login
Get assignments - worked fine
Logout
Get assignments - still working after logout
This shouldn't be the expect behavior, right?

Yes and no :upside_down_face: The problem is that JWT itself is stateless. And if it's configured that TTL is 1h (default one currently). It doesn't really matter if you called logout (in our app) or not. As I know, there are only three ways to "workaround" this -> 
1. issue tokens with smaller TTL;
2. "fully manage" their usage on FE side (which we are currently doing). Meaning that FE decides if it "should or not store" the access token after calling logout url. Indeed, if FE will store the access token - it can still access resources while access token is valid. BUT, after it's passed its TTL -> it's completely dead, and cannot be renewed with refreshToken, since it would be cleaned from cache after logout.;
3. create some kind of cache of "invalidated" access tokens. But this will lead to growth of redis cache, which, IMO, is not a good way. Especially since we are already going with kinda reaaaaally big redis instances for prod usage :) 
4. 
==============
So, until "further discussion" (though not sure if it's really needed) -> this behaviour would be unchanged. That's what we've lost (full logout functionality) with JWT, while getting all the good sides of it

Reply:
I'm sorry, I'm not 100% familiar with the JWT auth process, but seems we don't store the access token, we just check if it's valid on the fly. If this is the intended, we are fine, otherwise we could go on @Skyld suggestion.

Saw here that they don't invalidate the access token also:
https://community.auth0.com/t/invalidating-an-access-token-when-user-logs-out/48365

Maybe just remove it from Frontend is easier? If it's being done, are we fine?

Reply:
Sorry, seems we wrote together. I think we are fine

Reply:
Yeah, we are actually fine :) Meaning that it's intended by basic JWT auth implementation.
To invalidate access tokens - we need to store either all the access tokens (until they are valid), or the list of invalid access tokens (they will get into this list right after the logout call). In both cases - this would increase redis storage size.
So, as said -> until we decide to do this, we are using "the old model", meaning that FE (in our case - TT portal) would manage the "lifecycle" of the tokens

Reply:
@Skyld , please read last comments regarding the token invalidations :wink:

Reply:
Now I remember why we did it like this, it is intentional :) We didn't wan't to assert the access tokens against Redis in every request (this is the reason why we are not storing it in cache), it is not required at the moment. I think the TTL is also ok, what we need is the FE removing the access token after logging out the user.